common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -713,6 +718,10 @@ def _internal_depth_first(item_location, is_root):
             bulk_record = self._get_bulk_ops_record(location.course_key)
             bulk_record.dirty = True
             self.collection.remove({'_id': {'$in': to_be_deleted}})
+
+        # Now it's been published, add the object to the courseware search index so that it appears in search results
+        self.do_index(location)
```

Comment:
One challenge here will be that the triggering of do_index() is deep in the bowels on lib/xmodule/modulestore which is not "Django aware". So you probably won't be able to delegate to Celery at the level of the code stack. You might need to raise a Django signal and then have a handler in the CMS level of the stack to listen on the signal and schedule a celery task.

I think Celery integration should be a quick  follow-on story.